helm/lib/dependabot/helm/update_checker/latest_version_resolver.rb

```diff
@@ -0,0 +1,192 @@
+# typed: strong
+# frozen_string_literal: true
+
+require "dependabot/helm/package/package_details_fetcher"
+require "sorbet-runtime"
+require "dependabot/git_commit_checker"
+require "dependabot/helm/version"
+
+module Dependabot
+  module Helm
+    class LatestVersionResolver
+      extend T::Sig
+
+      DAY_IN_SECONDS = T.let(24 * 60 * 60, Integer)
+
+      sig do
+        params(
+          dependency: Dependabot::Dependency,
+          credentials: T::Array[Dependabot::Credential],
+          cooldown_options: T.nilable(Dependabot::Package::ReleaseCooldownOptions)
+        ).void
+      end
+      def initialize(dependency:, credentials:, cooldown_options:)
+        @dependency = dependency
+        @credentials = credentials
+        @cooldown_options = cooldown_options
+      end
+
+      sig { returns(Dependabot::Dependency) }
+      attr_reader :dependency
+
+      # To filter versions in cooldown period based on version tags from registry call
+      sig do
+        params(tags: T::Array[String], tags_with_release_date: T::Array[GitTagWithDetail])
+          .returns(T::Array[String])
+      end
+      def filter_versions_in_cooldown_period_using_oci(tags, tags_with_release_date)
+        select_tags_which_in_cooldown_using_oci(tags_with_release_date)&.each do |tag_name|
+          # Iterate through versions and filter out those matching the tag_name
+          tags.reject! do |version|
+            version == tag_name
+          end
+        end
+        tags
+      rescue StandardError => e
+        Dependabot.logger.error("Error filter_versions_in_cooldown_period_for_oci:: #{e.message}")
+        tags
+      end
```

Comment:
Instead of returning `tags` twice we can use an ensure block.


```suggestion
      def filter_versions_in_cooldown_period_using_oci(tags, tags_with_release_date)
        select_tags_which_in_cooldown_using_oci(tags_with_release_date)&.each do |tag_name|
          # Iterate through versions and filter out those matching the tag_name
          tags.reject! do |version|
            version == tag_name
          end
        end
      rescue StandardError => e
        Dependabot.logger.error("Error filter_versions_in_cooldown_period_for_oci:: #{e.message}")
      ensure
        tags
      end
```